R/compute_centiloid.R

```diff
@@ -0,0 +1,148 @@
+#' Compute Centiloid Value
+#'
+#' Computes centiloid values based on amyloid PET tracer, SUVR value and pipeline,
+#' and reference region.
+#' Also allows for custom formula parameters.
+#'
+#' @param tracer Amyloid PET tracer from ag_neuro dataset
+#'
+#'   A character string is expected.
+#'   When `custom_params` is not provided, accepted values are:
+#'   '18F-Florbetapir', '18F-Florbetaben'
+#'   If `custom_params` is provided, any string is accepted.
+#'
+#' @param pipeline SUVR pipeline from nv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When `custom_params` is not provided, accepted values are:
+#'   'AVID FBP SUVR PIPELINE', 'BERKELEY FBP SUVR PIPELINE', 'BERKELEY FBB SUVR PIPELINE'
+#'   If `custom_params` is provided, any string is accepted.
+#'
+#' @param ref_region Reference region from suppnv_neuro dataset
+#'
+#'   A character string is expected.
+#'   When `custom_params` is not provided, accepted values are:
+#'   'Whole Cerebellum', 'Composite Reference Region'
+#'   If `custom_params` is provided, any string is accepted.
+#'
+#' @param suvr SUVR value
+#'
+#'   A numeric value is expected.
+#'
+#' @param custom_params Optional list containing custom formula parameters 'm' and 'c'
+#'
+#'   A list with two named numeric values is expected when provided:
+#'   list(m = numeric_value, c = numeric_value)
+#'   When provided, this overrides the standard formula parameters
+#'
+#' @details
+#' The centiloid scale is a standardized quantitative measure for amyloid PET imaging
+#' that allows comparison between different tracers and analysis methods. This function
+#' converts SUVR values to the centiloid scale based on published conversion
+#' equations for specific tracer, pipeline, and reference region combinations.
+#'
+#' Centiloid is calculated as:
+#'
+#' Centiloid = m * SUVR + c
+#'
+#' where m and c are formula parameters. If `custom_params` is provided, this formula
+#' is used regardless of the tracer, pipeline, or reference region values.
+#'
+#' If a matching standard combination is not found and no `custom_params` are provided,
+#' a warning is issued and `NA_real_` is returned.
+#'
```

Comment:
Can you please in some way include references and formulas for each of the possilble combinations of `m` and `c` that are pre-specified in `tracer`, `pipeline`, `ref_region`?

Reply:
Thanks!